pkg/clusteragent/languagedetection/patcher_test.go

```diff
@@ -506,7 +525,25 @@ func TestRun(t *testing.T) {
 			},
 		}})
 
-	mockDeploymentEvent := workloadmeta.Event{
+	mockDeploymentEventToFail := workloadmeta.Event{
+		Type: workloadmeta.EventTypeSet,
+		Entity: &workloadmeta.KubernetesDeployment{
+			EntityID: workloadmeta.EntityID{
+				Kind: workloadmeta.KindKubernetesDeployment,
+				ID:   "test-namespace/" + longContNameDeploymentName,
+			},
+			DetectedLanguages: map[langUtil.Container]langUtil.LanguageSet{
+				*langUtil.NewContainer("some-cont"):            {"java": {}, "python": {}},
+				*langUtil.NewInitContainer("python-ruby-init"): {"ruby": {}, "python": {}},
+				// container name is 69
+				*langUtil.NewInitContainer("very-very-very-very-very-very-very-very-very-very-very-very-long-name"): {"ruby": {}, "python": {}},
```

Comment:
I think it would be better to use a function that generates a string of a given size.
That way the intention will be clearer.
I found this one: https://github.com/DataDog/datadog-agent/blob/ab5ed913c1bafff303122bded045be62077b6e46/test/benchmarks/util.go#L19 not sure if we have others. It might be good to define it in a place where it can be reused.

Reply:
I see your point, but using the method you are proposing introduces randomness in the unit test, which is not very good in my opinion, as it might cause flakiness (although I agree that in this specific case I don't think it will be flaky)

WDYT?

Reply:
In this particular case, the important thing is the size, the characters shouldn't cause flakiness.